src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -204,9 +221,9 @@ public FriendlyByteBuf.Reader<? extends CustomPacketPayload> getReader(ResourceL
                     final Connection connection = ConnectionUtils.getConnection(context);
                     final PacketListener listener = connection.getPacketListener();
                     if (listener instanceof ServerGamePacketListener serverListener) {
-                        serverListener.disconnect(Component.translatable("neoforge.network.invalid_flow", flow));
+                        serverListener.disconnect(Component.translatableWithFallback("neoforge.network.invalid_flow", "Failed to process a payload that was send with an invalid flow: %s", flow));
```

Comment:
Not a fan of the translation duplication. The server should have translations loaded, so we can query the lang on the server instead of hardcoding its value.

Reply:
Actually it does not.
We do not load the translations anymore on the server side, I also thought we did, but that is not the case anymore so this is currently the best we can do.

We already discussed internally that we should indeed refactor this and probably generate our en_us file using a data generator, so that this gets cleaned.